distrib_rl/Agents/MARLAgent.py

```diff
@@ -20,7 +20,7 @@ def __init__(self, cfg):
 
 
     @torch.no_grad()
-    def gather_timesteps(self, policy, env, num_timesteps=None, num_seconds=None, num_eps=None):
+    def gather_timesteps(self, policy, env, num_timesteps=None, num_seconds=None, num_eps=None, trajectory_callback=None):
```

Comment:
Perhaps this should be a generator instead of accepting a callback argument?

Reply:
If so, then we should also look at cleaning up the termination logic here (endless generator, let the caller decide when to stop requesting new experience)

Reply:
left termination logic as-is for now